src/libraries/MathLib.sol

```diff
@@ -2,11 +2,35 @@
 pragma solidity >=0.8.0;
 
 library MathLib {
+    function max(uint256[] memory values) internal pure returns (uint256) {
+        uint256 maxValue = values[0];
+        for (uint256 i = 1; i < values.length; i++) {
+            if (values[i] > maxValue) {
+                maxValue = values[i];
+            }
+        }
+        return maxValue;
+    }
+
     function max(uint256 a, uint256 b) internal pure returns (uint256) {
         return a > b ? a : b;
     }
 
+    function min(uint256[] memory values) internal pure returns (uint256) {
```

Comment:
except in situation where I don't know the exact length, I would not use such a functionality because this is way too much gas expensive